tests/test_mnist.py

```diff
@@ -111,19 +116,21 @@ def test_mnist_fedavg_mpi():
 
     num_clients = 2
     cfg = OmegaConf.structured(Config)
+    cfg.fed.args.num_local_epochs=2
     model = CNN(1, 10, 28)
+    loss_fn = torch.nn.CrossEntropyLoss()
     train_datasets, test_dataset = process_data(num_clients)
 
     if comm_size > 1:
         if comm_rank == 0:
-            rt.run_server(cfg, comm, model, num_clients, test_dataset, "test_mnist")
+            rm.run_server(cfg, comm, model, loss_fn, num_clients, test_dataset, "test_mnist")
         else:
-            rt.run_client(cfg, comm, model, num_clients, train_datasets)
+            rm.run_client(cfg, comm, model, loss_fn, num_clients, train_datasets)
     else:
         assert 0
 
 
-@pytest.mark.mpi(min_size=3)
+@pytest.mark.mpi(min_size=2)
```

Comment:
Is there any reason to set it to 2? I originally set it to 3, because it allows to have two processes for clients. With 2 processes, we have only one process for clients.

Reply:
Yes, it's because it seems there is a storage limitation to downloading the MNIST data 3 times.

With the MPI size 3, it was fine to run on the local machine. However, in the GitHub action environment, the test failed due to the downloaded dataset was not completed properly. The error case is [here](https://github.com/APPFL/APPFL/actions/runs/4810043927/jobs/8562659977)

I suspected the reason is the limitation of the storage size in the GitHub action environment. The storage limitation is 500MB. When I check the relative package size for APPFL was around 300MB, and the MNIST dataset size was 63MB. So, if we download 3 times of the dataset 3 times, it's too tight to fit on 500MB. Also, I believe I could miss some dependency sizes because it's calculated manually. Therefore, I reduced the MPI size to test and the test runs without the dataset completion issue.